src/main/java/persistence/sql/ddl/DefaultEntityManager.java

```diff
@@ -0,0 +1,194 @@
+
+        Object id = FieldUtils.getValue(field, entity);
+
+        if (id == null) {
+            Object valueOfId = insert(entity, Statement.RETURN_GENERATED_KEYS);
+
+            FieldUtils.setValue(field, entity, valueOfId);
+
+            // id 가져오기
+            putCache(clazz, valueOfId, entity);
+        } else {
+            if (find(clazz, id) == null) {
+                insert(entity, Statement.NO_GENERATED_KEYS);
+            } else {
+                update(entity);
+            }
+
+            putCache(clazz, id, entity);
+        }
+
+        return entity;
+    }
+
+    private Object insert(Object entity, int autoGeneratedKeys) {
+        String queryOfInsert = insertDMLGenerator.generate(entity);
+
+        try (PreparedStatement statement = connection.prepareStatement(queryOfInsert, autoGeneratedKeys)) {
+            statement.execute();
+
+            if (autoGeneratedKeys == PreparedStatement.RETURN_GENERATED_KEYS) {
+                ResultSet generatedKeys = statement.getGeneratedKeys();
+
+                generatedKeys.next();
+
+                return generatedKeys.getObject(1);
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+
+        return null;
+    }
+
+    private void update(Object entity) {
+        String query = updateDMLGenerator.generate(entity);
+
+        jdbcTemplate.execute(query);
+    }
+
+    @Override
+    public void remove(Object entity) {
+        Class<?> clazz = entity.getClass();
+        EntityTable entityTable = EntityTable.from(clazz);
+
+        Field field = entityTable.getFieldByIdColumn();
+
+        Object id = FieldUtils.getValue(field, entity);
+
+        removeCache(clazz, id);
+
+        String query = deleteDMLGenerator.generateDeleteById(entityTable, id);
+
+        jdbcTemplate.execute(query);
+    }
+
+    @NotNull
+    private <T> RowMapper<Object> toObject(Class<T> clazz, Object id, EntityTable entityTable) {
+        return resultSet -> {
+            T entity = newInstance(clazz);
+
+            entityTable.getAllColumnNames().forEach(it -> {
+                Field field = entityTable.getFieldByColumnName(it);
+
+                applyValue(resultSet, it, field, entity);
+            });
+
+            putCache(clazz, id, entity);
+
+            return entity;
+        };
+    }
+
+    private static <T> void applyValue(ResultSet resultSet, String it, Field field, T entity) {
+        try {
+            Object value = resultSet.getObject(it);
+
+            FieldUtils.setValue(field, entity, value);
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @NotNull
+    private <T> T newInstance(Class<T> clazz) {
+        T t = null;
+        try {
+            t = clazz.newInstance();
+        } catch (InstantiationException e) {
+            throw new RuntimeException(e);
+        } catch (IllegalAccessException e) {
+            throw new RuntimeException(e);
+        }
```

Comment:
```suggestion
        } catch (InstantiationException | IllegalAccessException e) {
            throw new RuntimeException(e);
        }
```
1. 별다른 처리가 없다면 따로 두지 않아도 될 것 같아요.
2. 이런 로직도 사실 `Class` 값을 `EntityManager`가 알기보다는 `EntityTable`에서 수행하도록해도 좋을 것 같아요.